packager/live_packager_export.cc

```diff
@@ -70,49 +83,65 @@ size_t livepackager_buf_size(LivePackagerBuffer_t buf) {
-                          const uint8_t* init,
-                          size_t init_len,
-                          const uint8_t* media,
-                          size_t media_len,
-                          LivePackagerBuffer_t dest) {
+LivePackagerStatus_t livepackager_package(LivePackager_t lp,
+                                          const uint8_t* init,
+                                          size_t init_len,
+                                          const uint8_t* media,
+                                          size_t media_len,
+                                          LivePackagerBuffer_t dest) {
   shaka::SegmentData input_init(init, init_len);
   shaka::SegmentData input_media(media, media_len);
-  return lp->inner->Package(input_init, input_media, *dest->inner).ok();
+  shaka::Status status =
+      lp->inner->Package(input_init, input_media, *dest->inner);
+  std::string status_message = status.ToString();
+  return new (std::nothrow)
+      LivePackagerStatus_s{status_message.c_str(), status.ok()};
 }
 
-bool livepackager_package_timedtext_init(LivePackager_t lp,
-                                         const uint8_t* seg,
-                                         size_t seg_len,
-                                         LivePackagerBuffer_t dest) {
+LivePackagerStatus_t livepackager_package_timedtext_init(
+    LivePackager_t lp,
+    const uint8_t* seg,
+    size_t seg_len,
+    LivePackagerBuffer_t dest) {
   shaka::SegmentData input_seg(seg, seg_len);
   shaka::FullSegmentBuffer out;
-  if (!lp->inner->PackageTimedText(input_seg, out).ok()) {
-    return false;
+  shaka::Status status = lp->inner->PackageTimedText(input_seg, out);
+  std::string status_message = status.ToString();
+  if (!status.ok()) {
+    return new (std::nothrow)
+        LivePackagerStatus_s{status_message.c_str(), status.ok()};
   }
 
   dest->inner->AppendData(out.InitSegmentData(), out.InitSegmentSize());
-  return true;
+  return new (std::nothrow)
+      LivePackagerStatus_s{status_message.c_str(), status.ok()};
 }
 
-bool livepackager_package_timedtext(LivePackager_t lp,
-                                    const uint8_t* seg,
-                                    size_t seg_len,
-                                    LivePackagerBuffer_t dest) {
+LivePackagerStatus_t livepackager_package_timedtext(LivePackager_t lp,
+                                                    const uint8_t* seg,
+                                                    size_t seg_len,
+                                                    LivePackagerBuffer_t dest) {
   shaka::SegmentData input_seg(seg, seg_len);
   shaka::FullSegmentBuffer out;
-  if (!lp->inner->PackageTimedText(input_seg, out).ok()) {
-    return false;
+  shaka::Status status = lp->inner->PackageTimedText(input_seg, out);
+  std::string status_message = status.ToString();
+  if (!status.ok()) {
+    return new (std::nothrow)
+        LivePackagerStatus_s{status_message.c_str(), status.ok()};
```

Comment:
This won't work - since `status_message` is a local variable it will be destroyed when it goes out of scope. The pointer inside of `LivePackagerStatus_s` could be pointing to garbage when that happens.  We'll need to make a copy the string, e.g., using `strdup()`.